### PR TITLE
Dev corner: toolshed slides

### PR DESCRIPTION
**Pull request by slegras**
Hi Björn,

We have re-built the entire Tool Shed course for the french version of the G4B (currently ongoing).
We've made the course so that it fits the new training repository you're managing.

Hope this will be of any help!

### CODE REVIEW

**Comment by bgruening** [comment]
@slegras I didn't know that anyone has such material. This is awesome! Thanks. Can you please rebase, I can not merge it yet.

File: Dev-Corner/slides/toolshed.html [hunk 1]
```diff
@@ -0,0 +1,801 @@
...
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### 3.2 Tool suite: relationship between required repositories
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ###  Tool suite
+                            <img src="../images/toolsuite.png" class="no-border" >
```
Comment (ValentinMarcon) [inline Dev-Corner/slides/toolshed.html]: @slegras 
We don't understand in the picture that this method is the wrong without the oral speech

Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: I'll change that! Thanks for your feedback!

File: Dev-Corner/slides/toolshed.html [hunk 2]
```diff
@@ -0,0 +1,801 @@
...
+                            - Select a Tool Shed
+
+                            <img src="../images/goToToolshed_ss.png" class="no-border" >
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Install a tool from the Tool Shed
+                            - Search your tool
+                            <img src="../images/searchTool_ss.png" class="no-border" >
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: The picture is missing

File: Dev-Corner/slides/toolshed.html [hunk 3]
```diff
@@ -0,0 +1,801 @@
...
+                            - Check
+                            <img src="../images/checkToolInstall_ss.png" class="no-border" >
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Manage installed tools
+                            - Admin > Manage installed tools
+                            <img src="../images/manageInstalledRep_ss.png" class="no-border" >
+                            <img src="../images/repStatus_ss.png" class="no-border" >
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: Those picture are missing

File: Dev-Corner/slides/toolshed.html [hunk 4]
```diff
@@ -0,0 +1,801 @@
...
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Manage installed tools
+                            - Click on the name of a tool
+                              - Manage and browse the repository
+                            <img src="../images/manageRep_ss.png" class="no-border" >
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: This picture is missing

File: Dev-Corner/slides/toolshed.html [hunk 5]
```diff
@@ -0,0 +1,801 @@
...
+                            ###  How to install a workflow?
+                            <img src="../images/workflowOK_ss.png" class="no-border" >
+                            - Done!
+                        </script>
+                    </section>
+                </section>
+                <section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ## 3. Share your tool on a Tool Shed
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: I had renaming this to "Tool Shed repository types" or something like that

File: Dev-Corner/slides/toolshed.html [hunk 6]
```diff
@@ -0,0 +1,801 @@
...
+                            ```
+                            <requirements>
+                              <requirement type="package" version="4.0">tandem_repeats_finder</requirement>
+                            </requirements>
+                            ```
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### 3.2 Tool suite: relationship between required repositories
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]:
```
### 3.2 Tool suite
#### relationship between required repositories
```

File: Dev-Corner/slides/toolshed.html (hunk 1 above)
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: Agreed

File: Dev-Corner/slides/toolshed.html [hunk 7]
```diff
@@ -0,0 +1,801 @@
...
+                            <?xml version="1.0"?>
+                            <repositories description="Pipeline phylogeny">
+                               <repository toolshed="http://testtoolshed.g2.bx.psu.edu"
+                                 name="fasta_to_phylip" owner="gandres" changeset_revision="a895633568" />
+                               <repository name="mafft" owner="gandres" />
+                               <repository name="phyml" owner="gandres" />
+                            […]
+                            </repositories>
+                            ```
+                            - « toolshed » and « changeset_revision » are optional **but they are completed during upload on the toolshed**
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: You can maybe remove `- « toolshed » and « changeset...`, you are dealing with 2 slides after

File: Dev-Corner/slides/toolshed.html [hunk 8]
```diff
@@ -0,0 +1,801 @@
...
+                            ###  Workflow
+                            <img src="../images/workflow.png" class="no-border" >
+                            - How to order tools ?
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ###  Workflow
+                            - In galaxy
+                            <img src="../images/galaxyWorkflowEx.png" class="no-border" >
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: It should be great to have the same tools as the previous slide mafft, trimal ...

File: Dev-Corner/slides/toolshed.html [hunk 9]
```diff
@@ -0,0 +1,801 @@
...
+                    <section data-markdown>
+                        <script type="text/template">
+                            ###  Workflow
+                            <img src="../images/workflowRepDep2.png" class="no-border" >
+                            - **Everything must be on the same toolshed**
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ## Tips: what about tools working with *.loc files or new datatypes
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]:
```
## Tips
### What about tools working with *.loc files or new datatypes
```

File: Dev-Corner/slides/toolshed.html [hunk 10]
```diff
@@ -0,0 +1,801 @@
...
+                            - \*.loc files and datatypes files should be provided as *.sample file in a `tool-data` directory.
+
+                            - tool-data/*.loc.sample files come along with `tool_data_table_conf.xml.sample`.
+                            Their content or the file themselves are put to the right location on tool set up.
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Tools with *.loc files or new datatypes
+                            <img src="../images/repWithLoc_ss.png" class="no-border" width="60%">
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: This picture is missing

File: Dev-Corner/slides/toolshed.html [hunk 11]
```diff
@@ -0,0 +1,801 @@
...
+                            | Tools           | `$name`                  | `stringle`, `bowtie2`    | -->
+                            - Planemo will determine the repository type based on these conventions
+                        </script>
+                    </section>
+										<section data-markdown >
+                        <script type="text/template">
+                            ### Creation
+                            - Tools and tool suite .shed.yml:
+                              - implicity define repositories for each individual tool in the directory and build a suite for those
+                              <img src="../images/planemoAutomaticSuite.png"  >
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: This picture is missing

File: Dev-Corner/slides/toolshed.html (hunk 4 above)
Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: fixed

File: Dev-Corner/slides/toolshed.html (hunk 6 above)
Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: ok

File: Dev-Corner/slides/toolshed.html (hunk 8 above)
Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: ok

File: Dev-Corner/slides/toolshed.html (hunk 9 above)
Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: ok

File: Dev-Corner/slides/toolshed.html (hunk 11 above)
Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: ok

File: Dev-Corner/slides/toolshed.html (hunk 1 above)
Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: ok

File: Dev-Corner/slides/toolshed.html (hunk 6 above)
Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: ok

File: Dev-Corner/slides/toolshed.html (hunk 7 above)
Comment (slegras) [inline Dev-Corner/slides/toolshed.html]: ok

File: Dev-Corner/slides/toolshed.html (hunk 5 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: done

**Comment by loraine-gueguen** [comment]
@slegras I have added general questions, objectives and key points. And added you and me as contributors.

**Comment by loraine-gueguen** [comment]
The new link that I have added in the Dev-Corner/readme.md file pointing to the new Tool Shed slides  fails, but will fail until the branch has been merged to master : `http://bgruening.github.io/training-material/Dev-Corner/slides/toolshed.html`
Do I remove this link or not ?

**Comment by bgruening** [comment]
@loraine-gueguen sorry for missing this. Do you need any help here?

**Comment by loraine-gueguen** [comment]
Not your fault, this branch is old: it was created before the change of path for reveal.js, and I missed it when I merged it with master 10 days ago.

File: Dev-Corner/slides/toolshed.html [hunk 12]
```diff
@@ -0,0 +1,861 @@
...
+
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Available Tool Sheds
+                            - [Galaxy community Tool Shed](https://toolshed.g2.bx.psu.edu/) -> Main Tool Shed
+                            - [Galaxy community test Tool Shed](https://testtoolshed.g2.bx.psu.edu/) -> Sandbox for testing
+                            - [List of public Tool Sheds](https://wiki.galaxyproject.org/PublicGalaxyServers#Public_ToolSheds)
+                            - [IFB Tool Shed](http://toolshed.france-bioinformatique.fr/)
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: I think we can remove the IFB TS.
Add it in the Public TS if needed and broadcast the URL to the French community if neeeded.
... if needed 😖

**Comment by loraine-gueguen** [comment]
@bgruening It should be fine now for the paths (reveal.js, css...). What should I do for the link in  Dev-Corner/readme.md ? See comment above.

File: Dev-Corner/slides/toolshed.html [hunk 13]
```diff
@@ -0,0 +1,861 @@
...
+                            conda_auto_install = True
+                            conda_auto_init = False
+                            tool_dependency_dir = ../tool_dependencies
+                            ```
+                              - in config/dependency_resolvers_conf.xml (be careful order matters):
+                            ```
+                            <dependency_resolvers>
+                                  <conda />
+                                  <conda versionless="true" />
+                                  <tool_shed_packages />
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: By experience, I have change my mind.
The order should remain inchanged.
You can choose if you want the tool_dependencies during the tool installation. If so, Galaxy will take it.
If not, Galaxy will search for a conda package.

If we change the order, Galaxy may find a conda package in the good version but unfortunatly, the wrapper is not ready for it and still need the dedicaded tool_dependency

File: Dev-Corner/slides/toolshed.html [hunk 14]
```diff
@@ -0,0 +1,861 @@
...
+                            <img src="../images/searchTool_ss.png" class="no-border" >
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Install a tool from the Tool Shed
+                            - <span style="font-size:70%">Types of repository</span>
+                              - <span style="font-size:55%">Tool (orange): contains tools (tool wrappers)</span>
+                              - <span style="font-size:55%">Tool dependencies (red): contains only the file tool\_dependencies.xml.
+                              !! tool\_dependencies.xml file is no longer supported because of the use of conda!
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: tool_dependencies are still supported but not recommended
The old tool_dependencies will be still understood by Galaxy.

File: Dev-Corner/slides/toolshed.html [hunk 15]
```diff
@@ -0,0 +1,861 @@
...
+                            <img src="../images/launchInstallTool_ss.png" class="no-border" >
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Install a tool from the Tool Shed
+                            - Install the tool
+                            <img src="../images/installTool_ss.png" class="no-border" width="80%">
+                            - <span style="font-size:70%"> Selecting Tool Shed AND conda will make Galaxy to install both</span>
+                            - <span style="font-size:70%"> Recommended: use conda </span>
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: @bgruening Should we add that there is currently an issue if you ran Galaxy through NFS?

File: Dev-Corner/slides/toolshed.html [hunk 16]
```diff
@@ -0,0 +1,861 @@
...
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Install a tool from the Tool Shed
+                            - Check
+                            <img src="../images/checkToolInstall_ss.png" class="no-border" >
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: One day, maybe refresh those screenshot to be package_* free

File: Dev-Corner/slides/toolshed.html [hunk 17]
```diff
@@ -0,0 +1,861 @@
...
+                            <img src="../images/manageInstalledRep_ss.png" class="no-border" >
+                            <img src="../images/repStatus_ss.png" class="no-border" >
+                        </script>
+                    </section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ### Manage installed tools
+                            - Click on the name of a tool
+                              - Manage and browse the repository
+                            <img src="../images/manageRep_ss.png" class="no-border" >
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: This picture should be refresh to show the last UI change to display the "Dependency Resolver Details" status

File: Dev-Corner/slides/toolshed.html [hunk 18]
```diff
@@ -0,0 +1,861 @@
...
+                            ```
+                            <span style="font-size:60%">On install, the content of the file and the right
+                            path to the *.loc file is added to `shed_tool_data_table_conf.xml`</span>
+                        </script>
+                    </section>
+                </section>
+                <section>
+                    <section data-markdown>
+                        <script type="text/template">
+                            ## 4. Publication with Planemo
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: Can you use this picture http://bgruening.github.io/training-material/Dev-Corner/slides/tool_integration.html#/6/3 to fit with the tool_integration part?

File: Dev-Corner/slides/toolshed.html [hunk 19]
```diff
@@ -0,0 +1,861 @@
...
+                            | Tools           | `$name`                  | `stringle`, `bowtie2`    | -->
+                            - Planemo will determine the repository type based on these conventions
+                        </script>
+                    </section>
+										<section data-markdown >
+                        <script type="text/template">
+                            ### Configure repository
+                            - Tools and tool suite .shed.yml:
+                              - implicity define repositories for each individual tool in the directory and build a suite for those
+                              <img src="../images/planemoAutomaticSuite.png"  >
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: I think that a simple .shed.yml should be interesting before one for suite

File: Dev-Corner/slides/toolshed.html [hunk 20]
```diff
@@ -0,0 +1,861 @@
...
+                            #### Step 2: Use planemo to publish your tool
+                            - `planemo config_init`
+                            - `planemo shed_init`
+                            - `planemo shed_lint`
+                            - `planemo shed_create`
+                        </script>
+                    </section>
+                    <section data-markdown >
+                        <script type="text/template">
+                            ## Answers
```
Comment (lecorguille) [inline Dev-Corner/slides/toolshed.html]: Should be move somewhere accessible for trainers?

File: Dev-Corner/slides/toolshed.html (hunk 12 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: removed

File: Dev-Corner/slides/toolshed.html (hunk 13 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: left default values

File: Dev-Corner/slides/toolshed.html (hunk 14 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: fixed

File: Dev-Corner/slides/toolshed.html (hunk 16 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: Not changed. I agree to refresh that later, but right now, I think that packages are still essential for most tools.

File: Dev-Corner/slides/toolshed.html (hunk 17 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: Done

File: Dev-Corner/slides/toolshed.html (hunk 18 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: Done.

File: Dev-Corner/slides/toolshed.html (hunk 19 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: I agree. Done

File: Dev-Corner/slides/toolshed.html (hunk 20 above)
Comment (loraine-gueguen) [inline Dev-Corner/slides/toolshed.html]: I have removed the "Answers" section. I have added some elements into `aside class="notes" ` tags. Should I create something somewhere else for trainees ?

File: Dev-Corner/slides/toolshed.html (hunk 15 above)
Comment (bgruening) [inline Dev-Corner/slides/toolshed.html]: I guess we can mention this in our talk or if people ask, no need to add it here.

**Comment by bgruening** [comment]
Great work! Thanks.

**Pull request merged**